### ISSUE TITLE
unarchive module add directory_mode parameter

### ISSUE THREAD

**Issue by edysli** [ansible/ansible-modules-core#3048]
##### Issue Type:

Feature Idea
##### Component Name:

unarchive
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Summary:

Currently, using unarchive with `mode=0644` creates directories without the execute permission, making them impossible to `cd` into. It would be nice to have a `directory_mode` parameter for unarchive, just like the copy module has, in order to set the proper mode on directories (`directory_mode=0755`).

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
I am all for this !

**Comment by ansibot** [comment]
@edysli we are unable to validate that "" is an existing ansible module.

We are expecting a component name that matches something that can be found in:
- http://docs.ansible.com/ansible/list_of_all_modules.html

Please revise the component name in the description.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by edysli** [comment]
@ansibot component name "unarchive" added to description

**Comment by dagwieers** [comment]
needs_contributor

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by edysli** [comment]
This issue was moved to ansible/ansible#19760
